matsim/src/main/java/org/matsim/core/population/algorithms/ChooseRandomLegModeForSubtour.java

```diff
@@ -119,13 +129,49 @@ public ChooseRandomLegModeForSubtour(
 	 * the modes should be available for such trips (e.g. car-sharing does not make much sense for such a trip),
 	 * thus the list of modes available for single-trip subtours can be specified independently. As mentioned,
 	 * it is initialized by the constructor to the full list of chain based modes.
-	 * 
+	 *
 	 * @param singleTripSubtourModes
 	 */
 	public void setSingleTripSubtourModes(final String[] singleTripSubtourModes) {
 		this.singleTripSubtourModes = Arrays.asList(singleTripSubtourModes);
 	}
-	
+
+	/**
+	 * Return all possible choices for a certain plan.
+	 */
+	public List<Candidate> determineChoiceSet(final Plan plan) {
+
+		final Id<? extends BasicLocation> homeLocation = ((Activity) plan.getPlanElements().get(0)).getFacilityId()!=null ?
+				((Activity) plan.getPlanElements().get(0)).getFacilityId() :
+				((Activity) plan.getPlanElements().get(0)).getLinkId();
+		Collection<String> permissibleModesForThisPlan = permissibleModesCalculator.getPermissibleModes(plan);
+
+		return determineChoiceSet(
+						homeLocation,
+						TripStructureUtils.getTrips(plan),
+						TripStructureUtils.getSubtours(plan),
+						permissibleModesForThisPlan);
+	}
+
+	/**
+	 * Return whether a subtour is mass conserving according to configuration.
+	 */
+	public boolean isMassConserving(final Subtour subtour) {
+
+		// non-closed subtours without any parent do not necessarily violate mass conservation
+		if (behavior == SubtourModeChoice.Behavior.betweenAllLessConstraints &&
+				!subtour.isClosed() && subtour.getParent() == null) {
+			return true;
+		}
```

Comment:
This assumes that all non-closed root subtours are mass conserving, which I think, is not the case:

Here, the car stays parked at work but is later picked up at other, which makes the non-closed root subtour not mass-conserving. Is that deliberate? 
```
            S
           | |
           | |
        walk walk
           | |
           | |
H ---car--- W ---pt--- O ---car--- L
```


In this example, the mass-conserving property depends on the assumption, where vehicles of chain-based modes are initially parked. I suspect, that MATSim in general places chain-based vehicles at the first activity by default, so this should also not be mass-conserving?
```
           S
          | |
          | |
       walk walk
          | |
          | |
H ---pt--- W ---pt--- O ---car--- L
```

Reply:
That is true that your first example is not mass conserving. The intention was rather to allow subtour mode choice to change it anyway. Nevertheless, I have changed this method back to avoid any confusion in this regard.

The second example would depend on the assumptions, but the goal is to be less restrictive. If this is used together with the option to run change single trip mode with a certain probably the mass conservation will be violated anyway, which is fine for the intended use-case.